test/error-message.js

```diff
@@ -0,0 +1,64 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/error-message';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const emptyStringError = {
+	ruleId: 'error-message',
+	message: 'Error message should not be an empty string'
+};
+
+const noMessageError = {
+	ruleId: 'error-message',
+	message: 'Pass a message to the error object'
+};
+
+ruleTester.run('error-message', rule, {
+	valid: [
+		`throw new Error('error')`,
+		`throw new TypeError('error')`,
+		`throw new MyCustomError('error')`,
+		'throw new Error(lineNumber=2)',
+		'throw new Error([])',
+		'throw foo()',
+		'throw err',
+		'throw 1',
+		`const err = TypeError('error');
+		 throw err;`
```

Comment:
Can you do a test like this here:

```js
let err = new Error();
err = 1;
throw err;
```

Just to ensure the variable tracking works both ways.